iamclient/iamclient.go

```diff
@@ -179,12 +179,9 @@ func (client *Client) InstallCertificates(
 		certConfirmation := cloudprotocol.InstallCertData{Type: cert.Type, NodeID: cert.NodeID}
 
 		response, err := client.certificateService.ApplyCert(ctx, request)
-		if err == nil {
-			certConfirmation.Serial, err = certProvider.GetCertSerial(response.CertUrl)
-		}
-
 		if err == nil {
 			certConfirmation.Status = "installed"
+			certConfirmation.Serial = response.Serial
```

Comment:
I guess we should set serial in either case. To indicate which certificate failed.

Reply:
fixed